vividus-http-client/src/main/java/org/vividus/http/client/HttpResponse.java

```diff
@@ -114,18 +116,30 @@ public HttpResponse verifyStatusCode(int... codes)
         {
             StringBuilder builder = new StringBuilder("Service returned response with unexpected status code: [")
                     .append(statusCode).append("]. Expected code is one of: ");
-            throw new ExternalServiceException(appendWithSeparators(builder, codes, ", ").toString());
+            appendWithSeparators(builder, codes, ", ");
+            throw new ExternalServiceException(appendResponseBody(builder.toString()));
         }
         return this;
     }
 
+    private String appendResponseBody(String message)
+    {
+        return Stream.of(responseHeaders)
+                     .filter(h -> HttpHeaders.CONTENT_TYPE.equals(h.getName())
+                         && h.getValue().contains(ContentType.APPLICATION_JSON.getMimeType()))
+                     .findAny()
+                     .map(h -> getResponseBodyAsString())
+                     .map(body -> message + ". Response body: " + body)
```

Comment:
there is a high chance we'll explode our error messages, this behaviour should be configurable at HttpClient level